### ISSUE TITLE
Smoketesting storybook

### ISSUE THREAD

**Issue by hamishtaplin** [storybooks/storybook#1862]
Are there any mechanism provided, either at build-time or runtime to check that each story renders without errors? As part of our CI, we build storybook on Travis and then deploy to Github pages—if a story breaks somehow (easily done) then we end up deploying a broken storybook application.

Currently, we are running the storybook app on Travis and then using webdriver.io (Selenium) to navigate to each story and check for errors. However, due to the way storybook renders the menu, it's really hard to parse for "links" (there are none, just divs) to stories when using "nested" stories.

The closest thing I can see is using [storyshots](https://github.com/storybooks/storybook/tree/master/addons/storyshots) to render each story—is there anything else?

**Comment by Hypnosphi** [comment]
There is a `--smoke-test` option for `start-storybook` in `3.3.0-alpha`, but looks like it doesn't serve your needs: it just checks that compilation is successful

I'll look into how menu items are rendered and probably will add some data attributes

**Comment by hamishtaplin** [comment]
@Hypnosphi Awesome, that would be really helpful 👍

**Comment by Hypnosphi** [comment]
By the way, you can obtain the stories tree in runtime:
```js
import {getStorybook} from '@storybook/react`;

const stories = getStorybook();
```
And then form the links using `?selectedKind=<kind>&selectedStory=<story>` pattern

**Comment by Hypnosphi** [comment]
>  add some data attributes
In fact they are already present, it's `data-name`

**Comment by ndelangen** [comment]
@hamishtaplin I wonder what in storyshots made you look for something else.

Storyshots will assure you all stories render exactly as before. If anything changes, it will fail.
Sounds like that's what you wanted?

**Comment by tmeasday** [comment]
You can use storyshots to do exactly this, with the [renderOnly](https://github.com/storybooks/storybook/tree/master/addons/storyshots#renderonly) test body.
